test/test_invalid_headers.py

```diff
@@ -239,20 +241,33 @@ class TestFilter(object):
     ]
 
     hdr_validation_combos = [
-        h2.utilities.HeaderValidationFlags(is_client, is_trailer)
-        for is_client, is_trailer in [
-            (True, True),
-            (True, False),
-            (False, True),
-            (False, False)
-        ]
+        h2.utilities.HeaderValidationFlags(
+            is_client, is_trailer, is_response_header
+        )
+        for is_client, is_trailer, is_response_header in (
+            itertools.product([True, False], repeat=3)
+        )
     ]
 
     hdr_validation_no_trailers = [
-        h2.utilities.HeaderValidationFlags(is_client, is_trailer)
-        for is_client, is_trailer in [
-            (True, False),
-            (False, False)
+        h2.utilities.HeaderValidationFlags(
+            is_client, is_trailer, is_response_header
+        )
+        for is_client, is_trailer, is_response_header in [
+            (True, False, False),
+            (False, False, False),
+        ]
+    ]
```

Comment:
We can modify `hdr_validation_no_trailers` further since it was created **only** for use in `test_matching_authority_host_headers`. However, the only combination out of the whole matrix that will (or maybe should?) work for `test_matching_authority_host_headers` is: 
```python
(
  (is_client, is_trailer, is_response_header'), 
  (True, False, False)
)
```
So we lose any point of having this in the TestFilter class to begin with.